app/scripts/metamask-controller.js

```diff
@@ -4573,8 +4687,16 @@ export default class MetamaskController extends EventEmitter {
 
     engine.push(this.metamaskMiddleware);
 
+    // if useRequestQueue feature flag is enabled
+    // use provider fetched from registry for the particular domain
+
+    // otherwise do the same old thing
     // forward to metamask primary provider
-    engine.push(providerAsMiddleware(provider));
+    if (this.preferencesController.getUseRequestQueue() === true) {
+      engine.push(providerAsMiddleware(proxyProviderForDomain));
+    } else {
+      engine.push(providerAsMiddleware(provider));
+    }
```

Comment:
I believe you've already set proxyProviderForDomain to the correct provider on line 4404 and don't need to check the useRequestQueue flag again here

Reply:
ah yes thank you